app/controllers/users.js

```diff
@@ -50,7 +52,22 @@ var configureUser = function configureUser(user, done) {
             if (setUserCredErr) {
               done(setUserCredErr);
             } else {
-              done();
+              connectorSvc = new ConnectorService();
+              connectorSvc.getConfig(function onGetConfig(getConfigErr, connectorConfig) {
+                if (getConfigErr) {
+                  done(getConfigErr);
+                  return;
+                }
+                connectorConfig.fog.token = credentials.token;
+
+                connectorSvc.setFogToken(connectorConfig, function onSetFogToken(setFogTokenErr) {
```

Comment:
Couldn't we just pass `credentials.token` to the `setFogToken` function and move all the logic of reading current config and update the property to there?

Reply:
Done!